lib/jekyll-admin/server/collection.rb

```diff
@@ -47,26 +49,70 @@ def collection
         collection[1] if collection
       end
 
+      def request_path
+        sanitized_path File.join(site.source, request_payload["path"])
```

Comment:
This should just be `sanitized_path(request_paylod["path"])`. See https://github.com/jekyll/jekyll-admin/blob/140a1f9f598341700796f4044657ea6808f7ea10/lib/jekyll-admin/server.rb#L53-L55.